### PR TITLE
Fixed issue #18235: No control on response and token export on survey archive export

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

File: application/controllers/admin/Export.php [hunk 1]
```diff
@@ -957,64 +957,71 @@ public function exportMultipleSurveys(string $sSurveys, string $sExportType)
             if ($iSurveyID === false) {
                 continue;
             }
-            if (Permission::model()->hasSurveyPermission($iSurveyID, 'responses', 'export')) {
```
Comment (Shnoulle) [inline application/controllers/admin/Export.php:960]: I don't have issue with multiple ? 
It"s OK, only on single.

But seems you fix a lot more

File: application/controllers/admin/Export.php [hunk 2]
```diff
@@ -1054,6 +1061,13 @@ private function exportarchive(int $iSurveyID, bool $bSendToBrowser = true)
     {
         $survey = Survey::model()->findByPk($iSurveyID);
 
+        if (
+            ($survey->hasTokensTable && !Permission::model()->hasSurveyPermission($iSurveyID, 'tokens', 'export'))
+            || !Permission::model()->hasSurveyPermission($iSurveyID, 'responses', 'export')
+        ) {
+            throw new CHttpException(403, gT("You do not have permission to access this page."));
```
Comment (Shnoulle) [inline application/controllers/admin/Export.php:1068]: this is the reported issue ;)

**Review by Shnoulle — APPROVED** [review]

**Review by Shnoulle — APPROVED** [review]

**Comment by olleharstedt** [comment]
Weird. Did this break master CI after merge?

**Comment by olleharstedt** [comment]
│ PHPUnit\Framework\ExpectationFailedException: Title validation in question-title-warning are not empty on success, get “The question code is mandatory.”
   │ Failed asserting that two strings are equal.

**Comment by Shnoulle** [comment]
See : https://github.com/LimeSurvey/LimeSurvey/actions/runs/2641395872/attempts/1 

I just rerun to have it working … (9 minutes after)
https://github.com/LimeSurvey/LimeSurvey/actions/runs/2641395872/attempts/2

Maybe a timing issue ? Need to wait longer in the test ?

**Comment by olleharstedt** [comment]
I can try adding a longer sleep, I guess... :(

**Comment by Shnoulle** [comment]
I relaunch the test to check without updating anything.

I see @gabrieljenik add a 10 second time in one PR … i really think it's too long …

**Comment by olleharstedt** [comment]
10 second is not acceptable, no. Where...?

**Comment by Shnoulle** [comment]
https://github.com/LimeSurvey/LimeSurvey/pull/2489#pullrequestreview-1027557223

Else : no need 2 seconds : already pass : https://github.com/LimeSurvey/LimeSurvey/runs/7278683981 and 2 attempt

**Pull request merged**